src/overlays/effects/ovl_Effect_Ss_Hitmark/z_eff_ss_hitmark.c

```diff
@@ -1,27 +1,128 @@
 /*
  * File: z_eff_ss_hitmark.c
  * Overlay: ovl_Effect_Ss_Hitmark
- * Description:
+ * Description: Hit Marks
  */
 
 #include "z_eff_ss_hitmark.h"
+#include "objects/gameplay_keep/gameplay_keep.h"
+
+#define rTexIndex regs[0]
+#define rType regs[1]
+#define rPrimColorR regs[2]
+#define rPrimColorG regs[3]
+#define rPrimColorB regs[4]
+#define rEnvColorR regs[5]
+#define rEnvColorG regs[6]
+#define rEnvColorB regs[7]
+#define rScale regs[8]
 
 #define PARAMS ((EffectSsHitmarkInitParams*)initParamsx)
 
-s32 EffectSsHitmark_Init(PlayState* play, u32 index, EffectSs* this, void* initParamsx);
+u32 EffectSsHitmark_Init(PlayState* play, u32 index, EffectSs* this, void* initParamsx);
 void EffectSsHitmark_Update(PlayState* play, u32 index, EffectSs* this);
 void EffectSsHitmark_Draw(PlayState* play, u32 index, EffectSs* this);
 
-#if 0
+static Color_RGB8 sColors[] = {
+    { 255, 255, 255 }, { 255, 255, 0 }, { 255, 255, 255 }, { 255, 0, 0 },   { 255, 200, 100 }, { 200, 150, 0 },
+    { 150, 100, 0 },   { 100, 50, 0 },  { 255, 255, 255 }, { 255, 0, 0 },   { 255, 255, 0 },   { 255, 0, 0 },
+    { 255, 255, 255 }, { 0, 255, 200 }, { 255, 255, 255 }, { 150, 0, 255 },
+};
```

Comment:
```suggestion
static Color_RGB8 sColors[] = {
    { 255, 255, 255 }, { 255, 255, 0 }, // EFFECT_HITMARK_WHITE Init
    { 255, 255, 255 }, { 255, 0, 0 }, // EFFECT_HITMARK_WHITE Update
    { 255, 200, 100 }, { 200, 150, 0 }, // EFFECT_HITMARK_DUST Init
    { 150, 100, 0 },   { 100, 50, 0 },  // EFFECT_HITMARK_DUST Update
    { 255, 255, 255 }, { 255, 0, 0 },   // EFFECT_HITMARK_RED Init
    { 255, 255, 0 },   { 255, 0, 0 }, // EFFECT_HITMARK_RED Update
    { 255, 255, 255 }, { 0, 255, 200 },  // EFFECT_HITMARK_METAL Init
    { 255, 255, 255 }, { 150, 0, 255 }, // EFFECT_HITMARK_METAL Update
};
```
I think there's something we can do here to group these colors a bit better. They can be grouped in pairs (primColor and envColor), and by type, but also how they're used initially vs what they're set to in update. Do you know why there are different colors or how they work?

Reply:
To be honest haven't looked to much into it. Currently I kind of want to get all the effects decompiled into main to see what is the best way to document them, then properly sync them up with OoT as well (assuming no else has plans to).

Reply:
Currently most if not all documentation is just taken from OoT.